src/ocspdash/web/blueprints/api.py

```diff
@@ -30,40 +31,41 @@
 def register_location_key():
     """Register a public key for an invited location."""
     # TODO: error handling (what if no invite, what if duplicate name, etc.)
-    print(request.data)
     try:
         unverified_claims = jwt.get_unverified_claims(request.data)
-    except jwt.JWTError:
-        abort(HTTPStatus.BAD_REQUEST, 'malformed JWT')
+    except jwt.JWTError as e:
+        raise InvalidUsage(f'failed to decode JWT: {str(e)}')
 
     try:
         unverified_public_key = b64decode(unverified_claims['pk']).decode('utf-8')
-    except KeyError:
-        abort(HTTPStatus.BAD_REQUEST, "'pk' missing from claims")
-    except (binascii.Error, UnicodeError):
-        abort(HTTPStatus.BAD_REQUEST, "failed to decode 'pk' claim")
+    except KeyError as e:
+        raise InvalidUsage(f'missing claim: {str(e)}')
+    except binascii.Error as e:
+        raise InvalidUsage(f"failed to decode 'pk' claim: {str(e)}")
+    except UnicodeError as e:
+        raise InvalidUsage(f"failed to decode 'pk' claim: {e.reason}")
 
     try:
         claims = jwt.decode(request.data, unverified_public_key)
-    except JWTError:
-        abort(HTTPStatus.BAD_REQUEST, 'failed to validate JWT')
+    except JWTError as e:
+        raise InvalidUsage(f'failed to decode JWT: {str(e)}')
 
     try:
         public_key = claims['pk']
-    except KeyError:
-        abort(HTTPStatus.BAD_REQUEST, "'pk' misisng from claims")
+    except KeyError as e:
+        raise InvalidUsage(f'missing claim: {str(e)}')
 
     try:
         invite_token = b64decode(claims['token'])
-    except KeyError:
-        abort(HTTPStatus.BAD_REQUEST, "'token' missing from claims")
-    except binascii.Error:
-        abort(HTTPStatus.BAD_REQUEST, "failed to decode 'token' claim")
+    except KeyError as e:
+        raise InvalidUsage(f'missing claim: {str(e)}')
+    except binascii.Error as e:
+        raise InvalidUsage(f"failed to decode 'token' claim: {str(e)}")
 
     try:
         manager.process_location(invite_token, public_key)
-    except ValueError:
-        abort(HTTPStatus.BAD_REQUEST, 'bad invite or public key')
+    except ValueError as e:
+        raise InvalidUsage(f'failed to process invite: {str(e)}')
 
     return '', HTTPStatus.NO_CONTENT
```

Comment:
I still don't like returning no content... why don't you have it send back some information about the location that's been registered?

Reply:
201 Created specifies you should return the URL of the created entity but there is no URL for the location.

What would you want to return? The location name? I was originally just jsonifying the location, but there's some fields that probably shouldn't be returned and going down the road of to_json and safe_to_json screams bad to me.